tests/unit/templating/routeGenerator.spec.ts

```diff
@@ -96,5 +96,31 @@ describe('RouteGenerator', () => {
 
       expect(models).to.equal('./controllerWith.tsInPath');
     });
+
+    it('adds js for routes if esm is true', () => {
```

Comment:
Moving towards esm compat is a good idea, but while adding the extension is one piece here, there are other pieces we should address (exports, moving away from require in templates)

Reply:
In terms of specs, I'd like to add an ESM project in the tests (see [fixtures](https://github.com/lukeautry/tsoa/tree/master/tests/fixtures)) that has catches some basic regressions.

Reply:
oh yeah let's catch everything.
most of our production services are using tsoa, and I would love to move them to esm

Reply:
we need to think how we want the tests to run, we can't mix the runs of esm and non esm.
because we need to load the files differently.
(we need a package.json with `"type": "module"`)

Reply:
Yes, unlike the other fixtures, we can't just throw the ts files in there and keep everything the same.
It should have it's own
- tsconfig
- package.json
- files + import